packages/cli/src/oclif/hooks/getAppRegistrationFieldChoices.js

```diff
@@ -1,13 +1,30 @@
 const { callAPI } = require('../../utils/api');
+const { BASE_ENDPOINT, IS_TESTING } = require('../../constants');
 
 module.exports = async function (options) {
   // We only need to run this for the register command
   if (!options || !options.id || options.id !== 'register') {
```

Comment:
@eliangcs I was trying to fetch the enum field options dynamically and provide them to the `options` directive (outlined in the [flag options in oclif's docs](https://cdn.zappy.app/0d0706249ddc17164425e12c7ee0f9cc.png)) so that `zapier register --help` would show the available options for `audience`, `category` and `role`  and we wouldn't have to write our own validation, but I couldn't figure out how to provide these options in a static context (the flags are [expected in a static context](https://github.com/zapier/zapier-platform/blob/596f2efe887cb01e62559a75487034f5c76a0549/packages/cli/src/oclif/commands/register.js#L142)). This is also why I had to write [my own validation with `_validateEnumFlags`](https://github.com/zapier/zapier-platform/blob/596f2efe887cb01e62559a75487034f5c76a0549/packages/cli/src/oclif/commands/register.js#L26).

My thought was that using a hook would allow me to asynchronously fetch these options from our backend and provide them to `buildFlags`, but I was mistaken.

Do you know of any ways that we could accomplish this?

Reply:
Thanks for the additional context! Using a hook to fetch options dynamically makes sense. I was able to override `options` by doing something like this in the hook:

```js
if (!options || !options.id || options.id !== 'register') {
  return null;
}

let formFields;
try {
  formFields = await callAPI('/apps/fields-choices');
} catch (e) {
  this.error(
    'Unable to connect to Zapier API. Please check your connection and try again.'
  );
}

const cmd = options.config.findCommand('register');
if (cmd && cmd.flags) {
  if (cmd.flags.audience) {
    cmd.flags.audience.options = formFields.intention.map((x) => x.value);
  }
  if (cmd.flags.role) {
    cmd.flags.role.options = formFields.role.map((x) => x.value);
  }
  if (cmd.flags.category) {
    cmd.flags.category.options = formFields.app_category.map((x) => x.value);
  }
}
```

which results in the following help text:

```
$ zapier-dev register --help
Register a new integration in your account.

USAGE
  $ zapier register [TITLE]

ARGUMENTS
  TITLE  Your integrations's public title. Asked interactively if not present.

OPTIONS
  -D, --desc=desc
      A sentence describing your app in 140 characters or less, e.g. "Trello is a team collaboration tool to organize tasks and keep projects on track."

  -a, --audience=global|private
      Are you building a public or private integration? Public integration development receives priority support and launch promotion from the Zapier team.

  -c, --category=accounting|ads-conversion|analytics|app-builder|app-families|bookmarks|business-intelligence|calendar|call-tracking|cms|commerce|communication|cont
  acts|content-files|crm|customer-appreciation|customer-support|dashboards|databases|developer-tools|devices|documents|drip-emails|ecommerce|education|email|email-n
  ewsletters|event-management|fax|files|filters|fitness|forms|fundraising|gaming|hr|human-resources|images|internet-of-things|invoices|it-operations|it-operations-e
  ducation|lifestyle-entertainment|marketing|marketing-automation|news|notes|notifications|payment-processing|phone|printing|product-management|productivity|project
  -management|reviews|sales-crm|scheduling|security-identity|server-monitoring|signatures|social|social-marketing|spreadsheets|support|tax|team-chat|team-collaborat
  ion|time-tracking|todo-lists|transactional-email|transcription|url-shorten|video|video-calls|webinars|website-app-building
      How would you categorize your app? Choose the most appropriate option for your app's core features.

  -d, --debug
      Show extra debugging output.

  -r, --role=employee|contractor|user
      What is your relationship with the app you're integrating with Zapier?

  -s, --[no-]subscribe
      Get tips and recommendations about this integration along with our monthly newsletter that details the performance of your integration and the latest Zapier
      news.

  -u, --url=url
      The homepage URL of your app. This will also be used as the email domain that would allow team members to join your integration as a Collaborator.

DESCRIPTION
  After running this, you can run zapier push to build and upload your integration for use in the Zapier editor.

  This will change the ./.zapierapprc (which identifies this directory as holding code for a specific integration).

EXAMPLES
  zapier register
  zapier register "My Cool Integration"
```

Reply:
> I will set these 3 flags to the flag.string type and leave _validateEnumFlags in place for validation. Are you okay with this?

Yes! I also think this is the best option we have. Thanks for testing!